src/zenml/materializers/built_in_materializer.py

```diff
@@ -26,6 +26,9 @@
 class BuiltInMaterializer(BaseMaterializer):
     """Read/Write JSON files."""
 
+    # TODO [LOW]: consider adding typing.Dict and typing.List
```

Comment:
Are they really the `correct` way of doing it? I thought this was a thing that was required for backwards compatibility reasons

Reply:
I think the more abstract`typing.Mapping` is reccomended for function parameters (generic key value map is accepted) but `typing.Dict` as a more specific annotation for the return type